locksmithctl/locksmithctl.go

```diff
@@ -216,25 +218,33 @@ func getClient() (*lock.EtcdLockClient, error) {
 		transport.TLSClientConfig = tlsconf
 	}
 
-	cfg := client.Config{
-		Endpoints: globalFlags.Endpoints,
-		Transport: transport,
-		Username:  globalFlags.EtcdUsername,
-		Password:  globalFlags.EtcdPassword,
-	}
+	for _, ep := range globalFlags.Endpoints {
+		cfg := client.Config{
+			Endpoints: []string{ep},
+			Transport: transport,
+			Username:  globalFlags.EtcdUsername,
+			Password:  globalFlags.EtcdPassword,
+		}
 
-	ec, err := client.New(cfg)
-	if err != nil {
-		return nil, err
-	}
+		ec, err := client.New(cfg)
+		if err != nil {
+			return nil, fmt.Errorf("unable to create etcd client: %w", err)
+		}
 
-	kapi := client.NewKeysAPI(ec)
+		kapi := client.NewKeysAPI(ec)
 
-	lc, err := lock.NewEtcdLockClient(kapi, globalFlags.Group)
-	if err != nil {
-		return nil, err
+		lc, err := lock.NewEtcdLockClient(kapi, globalFlags.Group)
+		if err != nil {
+			if errors.Is(err, syscall.ECONNREFUSED) {
+				continue
+			}
+			return nil, fmt.Errorf("unable to create etcd lock client: %w", err)
+		}
+
+		return lc, nil
 	}
-	return lc, err
+
+	return nil, errors.New("unable to create an etcd client")
```

Comment:
It should be fine to use `fmt.Errorf` here, even though we don't wrap anything, to avoid importing `errors` :smile: 
```suggestion
	return nil, fmt.Errorf("unable to create an etcd client")
```

Reply:
Yeah for sure... I just don't know why there is no `fmt.Error` only :D

Reply:
I just checked, we actually import the `errors` to assert the errors is a `connection refused` one (https://github.com/kinvolk/locksmith/commit/b37d41de9e44e84a3d69e3b78769de75e15d87a7#diff-7b29e35460d650d4b39ff1302e2c54d5817c848ab7a497d19569e2329fbb8cdaR238)